activerecord/lib/active_record/connection_adapters/postgresql/schema_statements.rb

```diff
@@ -5,10 +5,13 @@ class SchemaCreation < AbstractAdapter::SchemaCreation
         private
 
         def visit_ColumnDefinition(o)
-          o.sql_type = type_to_sql(o.type, o.limit, o.precision, o.scale)
-          o.sql_type << '[]' if o.array
+          o.sql_type = type_to_sql(o.type, o.limit, o.precision, o.scale, o.array)
           super
         end
+
+        def type_to_sql(type, limit, precision, scale, array)
+          @conn.type_to_sql(type, limit, precision, scale, array)
```

Comment:
Added the method in the subclass of `SchemaCreation` for the explicit arguments.